### ISSUE TITLE
VSCode Debugging.

### ISSUE THREAD

**Issue by ghost** [lluachaturbate/CBADK#1]
Is it posible to integrate the debugging part of this app with VSCode?

**Comment by lluachaturbate** [comment]
Unfortunately this would mean a major rework and make it harder to maintain, which probably would be overkill for this "small" project.
I will leave the issue open and if a lot of people want VSCode i'll give it a second thought.

**Comment by ghost** [comment]
I'd implement this myself but never worked with QT and just to download
10gb of stuff ( this is what the installer asked me ) for such a small
project it's an overkill

Googling around I fount this
https://code.visualstudio.com/Docs/extensions/example-debuggers
Il 02 apr 2016 11:44 PM, "lluachaturbate" notifications@github.com ha
scritto:

**Comment by ghost** [comment]
im a total noob at qt how is this debugger implemented? also what is the minimal package i need to open the app in QT?

**Comment by lluachaturbate** [comment]
Qt is just a framework for c++, so you can use any IDE you want, but need the qmake tool as an additional precompile step.
The easiest is to just get the online installer for your platform, it automatically sets you up with all you need and ready to go.

In regards of the debugger:
Qt has it's own script module, which wraps around the JS engine. You can either choose to install the sources in the installer or you can just download them as a single archive.
Unless you are advanced in c++ you probably won't understand a thing in there tho.

**Comment by ghost** [comment]
i know what QT is and i just wanted to avoid a massive install of it, also as i undestrand is no longer free/OSS?

**Comment by lluachaturbate** [comment]
Nope, it is and will stay available under (L)GPL. They have some extra tools (like device creation) that are only available with a commercial license, but the main framework is FOSS.

**Comment by ghost** [comment]
my cpp is really rusty but will try to give a look into it, would u be up to collaborating on it?

**Comment by lluachaturbate** [comment]
Of course, but as i said this is probably a huge task due to implementation details of the script engine.
Debugging shouldn't play such a big role for small JS Apps. So usually you should be fine with any debugger. Is there a particular reason you desperately want VSCode for?

**Comment by ghost** [comment]
The thing is having this kind of app model and really low support from CB
having a debugger like that would be awesome
Il 04 apr 2016 3:18 AM, "lluachaturbate" notifications@github.com ha
scritto:

**Comment by lluachaturbate** [comment]
Well the CB API doesn't have the best documentation and the appengine some serious flaws.
Sadly they doesn't answer to any dev related emails either (or they just don't like me?).
I guess they did the (sadly typical) mistake of ending their coders contract after the site and servers where set up. That there is no update to the engine since 2014 at least speaks to it. If you look at [the roadmap](https://chaturbate.com/apps/docs/roadmap.html#make-a-suggestion) there are some really easy to implement things on it, so i doubt there is anyone assigned to it.

There is no way to debug their engine anyway (I don't even know what kind of engine they are actually running). I tried to copy it's behavior as good as i can, but there is never a guarantee. 

The ADK is to have an easy and comfortable time developing and testing, because the process of uploading and starting apps on the testbed consumes so much time and effort that it renders it barely useless. It's only good when you pretty much sure your app will work and want to do a final test before letting people use it. And even then it's really complicated to create input (user actions).

But it doesn't seem like there is so much need for an ADK, at least judging by the number of feature requests i received. So far you are the only one and you spotted simple QOL features, thanks.

It just seems that barely nobody creates new apps since there are already well known ones out there. 

I'm happy to build additional (and complicated) features like remote debugging etc. when there is "a need" for it. But please don't take it personally that I won't code for days when there is barely no one using it. My time is just limited and it's not like anybody would pay me for this so i can take time off of work. 

CB didn't even took the time to change [their site](https://chaturbate.com/apps/docs/appdevkit.html) stating that there are alternative to the outdated (since 3years) first ADK someone made. It doesn't feel like they want to encourage users to make new apps.

**Comment by masterchief36** [comment]
I tried many times, and to be honest support on that site in general sucks
Il 04 apr 2016 6:11 AM, "lluachaturbate" <notifications@github.com> ha
scritto:

**Comment by ghost** [comment]
To be honest the support in that site is just terrible, lets not talk about how the dev side of it is managed. 

Im really greatfull that someone like you took the time to create something like this, i did my own part into this trying to standardize some of the core functionality in the apps and creating a framework on it (My repo [Chat.Apps](https://github.com/masterchief36/Chat.Apps) contains all my work)

As in for the request of complicated featues im not expecting anyone to lose so much time on stuff like that, again im a noob at the tecnology you are using, i would have loved to implement it myself and share it.

**Comment by lluachaturbate** [comment]
I looked a bit into VSCode debugging.
IMHO they are just stupid by not putting up a documentation for their debug protocol, but instead make it a typescript file.
That just makes it hard to work with it for people having nothing to do with weblanguages.
Also VSCode only communicates through stdin/stdout, which is not really suitable if you want to debug something your application loads.
So what we would have to do is a platform independent executable that pipes stdin/out through a local socket to communicate with the application.
And i have to write a new debugger and implement the protocol.

**Comment by ghost** [comment]
Well if u are willing to do it i can do the typescript part
Il 05 apr 2016 12:50 PM, "lluachaturbate" notifications@github.com ha
scritto:
